internal/factsengine/gatherers/packageversion.go

```diff
@@ -50,19 +54,85 @@ func (g *PackageVersionGatherer) Gather(factsRequests []entities.FactRequest) ([
 			continue
 		}
 
-		version, err := g.executor.Exec(
-			"rpm", "-q", "--qf", "%{VERSION}", factReq.Argument)
+		packageName := factReq.Argument
+		requestedVersion := ""
+		if strings.Contains(factReq.Argument, ",") {
+			arguments := strings.SplitN(factReq.Argument, ",", 2)
+			packageName = arguments[0]
+			requestedVersion = arguments[1]
+		}
+
+		version, err := executeRpmVersionRetrieveCommand(g.executor, packageName)
 		if err != nil {
-			gatheringError := PackageVersionCommandError.Wrap(factReq.Argument)
+			gatheringError := PackageVersionCommandError.Wrap(packageName)
 			log.Error(gatheringError)
 			fact = entities.NewFactGatheredWithError(factReq, gatheringError)
-		} else {
-			fact = entities.NewFactGatheredWithRequest(factReq, &entities.FactValueString{Value: (string(version))})
+			facts = append(facts, fact)
+			continue
 		}
 
+		if requestedVersion != "" {
+			installedVersion, err := executeRpmVersionRetrieveCommand(g.executor, packageName)
```

Comment:
We don't need to run this again. You already have the version in `version`.
This way you can remove the next error checking as well